feat(plugin-lighthouse): add lighthouse runner (#549)
code-pushup/cli: CONTRIBUTING.md

npm install
```

## Environment Variables

This table provides a quick overview of the environmental setup, with detailed explanations in the corresponding sections.

| Feature                          | Local Default | CI Default         | Description                                                                                                                   |
| -------------------------------- | ------------- | ------------------ | ----------------------------------------------------------------------------------------------------------------------------- |
| `env.INCLUDE_SLOW_TESTS` **❗️** | `false`       | `true`             | Controls inclusion of long-running tests. Overridden by setting. Details in the [Testing](#Testing) section.                  |

> | `env.CUSTOM_CHROM_PATH`          | N/A           | Windows **❗️❗️** | Path to Chrome executable. See [plugin-lighthouse/CONTRIBUTING.md](./packages/plugin-lighthouse/CONTRIBUTING.md#chrome-path). |

Reviewer: Typo.
```suggestion
| `env.CUSTOM_CHROME_PATH`          | N/A           | Windows **❗️❗️** | Path to Chrome executable. See [plugin-lighthouse/CONTRIBUTING.md](./packages/plugin-lighthouse/CONTRIBUTING.md#chrome-path). |
```

> | `env.CUSTOM_CHROM_PATH`          | N/A           | Windows **❗️❗️** | Path to Chrome executable. See [plugin-lighthouse/CONTRIBUTING.md](./packages/plugin-lighthouse/CONTRIBUTING.md#chrome-path). |

Reviewer: 👨‍💻 DX: It is not clear from this that the Chrome dependency is only required for using/contributing to the Lighthouse plugin, not the CLI as a whole.

Author: The path is needed for all plugin users. In general there are no "unreolved" problems in setup and usage. But as we saw there are. I would add a bit of text to the plugin readme to clarify the setup ad link to the trouble shooting section.

Reviewer: If I install CLI and only use ESLint and coverage plugins, I should not need to worry about any Chrome path. This dependency is (I assume) tied to Lighthouse only at the moment. Am I missing something?

Author: > If I install CLI and only use ESLint and coverage plugins, I should not need to worry about any Chrome path.
Yes. therefore in a `CONTRIBUTING.md`
> This dependency is (I assume) tied to Lighthouse only at the moment.
Yes therefor it is inside the `package.json` of `plugin-lighthouse`

> Am I missing something?
I guess no. But what should I change now?
It is inside a contributor related file. I added lighthouse only. Hope that fixes it.

Reviewer: My point was to explicitly add to the root `CONTRIBUTING` that that environment variable is only applicable when using or contributing to **Lighthouse** plugin. I unfortunately cannot see any "Lighthouse only" but I can see a link to CONTRIBUTING of Lighthouse, could be enough.

npx nx code-pushup -- collect
```

### Testing

> To skip long-running tests on your local environment set the following environment variable `SKIP_LONG_TESTS=true`.

Reviewer: I think the flag should work in the opposite way - as a contributor I should not be required to know about and set the flag if I want a smooth DX - in other words, one should opt in, not opt out.

Author: That way we would have to strictly configure it. I would rather prefer to have our CI run by default all tests instead of  a custom configuration that lives in the `action.yml` file.

Is there a way to detect if we are on a local setup?

Reviewer 2: > Is there a way to detect if we are on a local setup?

The best heuristic for that is to check if `CI` environment variable is defined.

Author: I could use that.. but then i would mess around with that var locally. 

Should I:
- A) keep as is
- B) refactor to derive the skip decision form `CI` (which would mean i have to set it to true locally if I want to run all tests.. Naming is not really intuitive.)

Reviewer 2: Let's first agree on what the desired outcome is :sweat_smile: 

1. Should long-running tests run in _CI_?
2. Should long-running tests run _locally by default_?

Current implementation assume the answer to both is _yes_, but @Tlacenka is suggesting something different (at least for local, not sure about CI).

As far as I'm concerned, I'm happy to go with the current implementation and tweak it later if we find it impractical. What do you guys think?

Regarding the implementation, we can create a helper function in `test-utils`, which could check multiple environment variables if needed.

Author: As a compromise I can implement the helper already and Go with `C`I || `SKIP_LONG_TESTS`, and we polish later.

Author: Updated the code with the helper function

Author: Isn't the code already skipping the long tests locally?

npx nx code-pushup -- collect
```

> ### Testing

Reviewer: 📖 Documentation hierarchy: I would not nest testing under development.

```suggestion
## Testing
```

> Long-running tests are treated based on the environment to achieve better DX.

Reviewer: ```suggestion
Some of the plugins have a longer runtime. In order to ensure better DX, longer tests are excluded by default when executing tests locally.
```

You can control the execution of long-running tests over the `INCLUDE_SLOW_TESTS` environment variable.

> To change this setup got you your root folder and open your `.env` file.

Reviewer: 🔡
```suggestion
To change this setup, open (or create) the `.env` file in the root folder.
```
